projects/opendr_ws_2/src/data_generation/data_generation/synthetic_facial_generation.py

```diff
@@ -0,0 +1,166 @@
+        self.parser.add_argument('--dump_2d_img', default='true', type=str2bool, help='whether to save 3d rendered image')
+        self.parser.add_argument('--dump_param', default='true', type=str2bool, help='whether to save param')
+        self.parser.add_argument('--dump_lmk', default='true', type=str2bool, help='whether to save landmarks')
+        self.parser.add_argument('--save_dir', default='./algorithm/DDFA/results', type=str, help='dir to save result')
+        self.parser.add_argument('--save_lmk_dir', default='./example', type=str, help='dir to save landmark result')
+        self.parser.add_argument('--img_list', default='./txt_name_batch.txt', type=str, help='test image list file')
+        self.parser.add_argument('--rank', default=0, type=int, help='used when parallel run')
+        self.parser.add_argument('--world_size', default=1, type=int, help='used when parallel run')
+        self.parser.add_argument('--resume_idx', default=0, type=int)
+        self.args = self.parser.parse_args()
+        self.synthetic = MultiviewDataGeneration(self.args)
+
+    def callback(self, data):
+        """
+        Callback that process the input data and publishes to the corresponding topics
+        :param data: input message
+        :type data: sensor_msgs.msg.Image
+        """
+
+        # Convert sensor_msgs.msg.Image into OpenDR Image
+
+        image = self.bridge.from_ros_image(data)
+        self.ID = self.ID + 1
+        # Get an OpenCV image back
+        image = np.float32(image.numpy())
+        name = str(f"{self.ID:02d}"+"_single.jpg")
+        cv2.imwrite(os.path.join(self.args.path_in, name), image)
+
+        if (self.ID == 5):
+            # Run SyntheticDataGeneration
+            self.synthetic.eval()
+            self.ID = 0
+            # Annotate image and publish results
+            current_directory_path = os.path.join(self.args.save_path, str("/Documents_orig/"))
+            for file in os.listdir(current_directory_path):
+                name, ext = os.path.splitext(file)
+                if ext == ".jpg":
+                    image_file_savepath = os.path.join(current_directory_path, file)
+                    cv_image = cv2.imread(image_file_savepath)
+                    cv_image = cv2.cvtColor(cv_image, cv2.COLOR_BGR2RGB)
+                    if self.image_publisher is not None:
+                        image = Image(np.array(cv_image, dtype=np.uint8))
+                        message = self.bridge.to_ros_image(image, encoding="bgr8")
+                        self.image_publisher.publish(message)
+        for f in os.listdir(self.args.path_in):
+            os.remove(os.path.join(self.args.path_in, f))
+
+
+def main(args=None):
+    rclpy.init(args=args)
+    try:
+        if torch.cuda.is_available():
+            print("GPU found.")
+            device = 'cuda'
+        else:
+            print("GPU not found. Using CPU instead.")
+            device = 'cpu'
+    except:
+        device = 'cpu'
```

Comment:
As commented earlier, the argparse code should be here [similar to other ROS nodes](https://github.com/opendr-eu/opendr/blob/b9dab8c0764143d9817944851defa8502750a146/projects/opendr_ws_2/src/opendr_perception/opendr_perception/face_recognition_node.py#L139-L175). Also the device checks should be updated to be the same as the ones in the linked node.